jupyterhub_singleuser_profiles/service.py

```diff
@@ -37,83 +39,54 @@ def __init__(self, server_url, token, namespace=None, verify_ssl=True):
     self.k8s_api_instance = client.CoreV1Api(k8s_client)
     self.os_client = DynamicClient(k8s_client)
 
-  def get_template(self, name):
-    tmplt = self.os_client.resources.get(api_version='template.openshift.io/v1', kind='Template')
-    if isinstance(tmplt, list):
-      tmplt = tmplt[0] # Openshift returns list of 2 resource types, we want template.openshift.io/v1/processedtemplates
-
+  def get_template(self, name, path):
+    confm = self.os_client.resources.get(api_version='v1', kind='ConfigMap')
     try:
-      response = tmplt.get(
+      response = confm.get(
           namespace=self.namespace,
           name=name
       )
-      template = response.to_dict()
+      confm = response.to_dict()
     except Exception as e:
       _LOGGER.error("Error: %s %s" % (name, e))
       template = None
+    template = yaml.load(confm['data'].get(path))
 
     return template
 
-  def process_template(self, user, template, **parameters):
-
-    self._set_template_parameters(
-      template,
-      USERNAME=escape(user),
-      **parameters
-    )
-    template["apiVersion"] = "template.openshift.io/v1"
-
-    endpoint = "{}/apis/template.openshift.io/v1/namespaces/{}/processedtemplates".format(
-        self.server_url,
-        self.namespace
-    )
-    response = requests.post(
-        endpoint,
-        json=template,
-        headers={
-            'Authorization': 'Bearer {}'.format(self.token),
-            'Content-Type': 'application/json'
-        },
-        verify=False #FIXME?
-    )
-    _LOGGER.debug("OpenShift master response template (%d): %r", response.status_code, response.text)
+  def process_template(self, user, service_name, template, configuration, labels=None):
 
-    try:
-        response.raise_for_status()
-    except Exception:
-        _LOGGER.error("Failed to process template: %s", response.text)
-        raise
+    tmp = jinja2.Template(json.dumps(template))
 
-    processed = response.json()
-    if len(processed["objects"]) != 1:
-      raise Exception("Template must contain a single ConfigMap or CustomResource")
+    result = tmp.render(parameters=configuration)
```

Comment:
AFAIK if you call `render` as 

```
tmp.render(configuration)
```

You will not have to user `paramteres` in the template (e.g. `parameters.worker_instances` vs. `worker_instances`), so the template variables will be shorter and directly mapped to `configuration` dict. 

Can you try this?